docs/development.rst

```diff
@@ -50,19 +50,277 @@ GitHub workflow
+the rebasing, and if successful will open another editor screen that
+lets you edit the text of the commit message::
+
+    # This is a combination of 5 commits.
+    # This is the 1st commit message:
+
+    Add documentation for rebasing.
+
+    # This is the commit message #2:
+
+    Please work, CI!
+
+    # This is the commit message #3:
+
+    Please work, CI!
+
+    # This is the commit message #4:
+
+    PLEASE work, CI!
+
+    # This is the commit message #5:
+
+    DONE!!!
+
+    # Please enter the commit message for your changes. Lines starting
+    # with '#' will be ignored, and an empty message aborts the commit.
+    #
+    # Date:      Tue Mar 5 17:00:39 2019 +0000
+    #
+    # interactive rebase in progress; onto 20fb835
+    # Last commands done (5 commands done):
+    #    squash c9c4a28 PLEASE work, CI!
+    #    squash 97a9458 DONE!!!
+    # No commands remaining.
+    # You are currently rebasing branch 'better-rebase-docs' on '20fb835'.
+    #
+    # Changes to be committed:
+    #       modified:   docs/development.rst
+    #
+    #
+
+We don't care about the commit messages for the squashed commits, so we
+delete them ending up with::
+
+    Add documentation for rebasing.
+
+    # Please enter the commit message for your changes. Lines starting
+    # with '#' will be ignored, and an empty message aborts the commit.
+    #
+    # Date:      Tue Mar 5 17:00:39 2019 +0000
+    #
+    # interactive rebase in progress; onto 20fb835
+    # Last commands done (5 commands done):
+    #    squash c9c4a28 PLEASE work, CI!
+    #    squash 97a9458 DONE!!!
+    # No commands remaining.
+    # You are currently rebasing branch 'better-rebase-docs' on '20fb835'.
+    #
+    # Changes to be committed:
+    #       modified:   docs/development.rst
+
+After saving and closing this editor session, we then get something like::
+
+    [detached HEAD 6b8a2a5] Add documentation for rebasing.
+    Date: Tue Mar 5 17:00:39 2019 +0000
+    1 file changed, 2 insertions(+), 2 deletions(-)
+    Successfully rebased and updated refs/heads/better-rebase-docs.
+
+Finally, after a successful rebase, you **must force-push**! If you try to
+push without specifying ``-f``, you will get a very confusing and misleading
+message::
+
+    $ git push origin better-rebase-docs
+    To github.com:mcveanlab/stdpopsim.git
+    ! [rejected]        better-rebase-docs -> better-rebase-docs (non-fast-forward)
+    error: failed to push some refs to 'git@github.com:mcveanlab/stdpopsim.git'
+    hint: Updates were rejected because the tip of your current branch is behind
+    hint: its remote counterpart. Integrate the remote changes (e.g.
+    hint: 'git pull ...') before pushing again.
+    hint: See the 'Note about fast-forwards' in 'git push --help' for details.
+
+**DO NOT LISTEN TO GIT IN THIS CASE!** Git is giving you is **terrible advice**
+which will mess up your branch. Instead, force-push::
+
+    $ git push -f origin better-rebase-docs
```

Comment:
Here I might make it clear that you should push to the branch that you are working from (i.e. not master). Might be able to add a reference to "topic_branch_name" as referenced above in  `git checkout -b topic_branch_name`

Reply:
and also make extra clear you are force-pushing to *your* fork.

Reply:
Good feedback thanks, will do.